R/driver-snowflake.R

```diff
@@ -5,6 +5,60 @@ NULL
 #' @rdname DBI-classes
 setClass("Snowflake", contains = "OdbcConnection")
 
+#' @description
+#' ## `odbcConnectionColumns()`
+#'
+#' If the catalog, or the schema arguments are NULL, attempt to infer
+#' by querying for CURRENT_DATABASE() and CURRENT_SCHEMA().  We do this
+#' to aid with performance, as the SQLColumns method is more performant
+#' when restricted to a particular DB/schema.
+#' @inheritParams DBI::dbListFields
+#' @rdname Snowflake
+#' @usage NULL
+setMethod("odbcConnectionColumns_", c("Snowflake", "character"),
+  function(conn,
+           name,
+           ...,
+           catalog_name = NULL,
+           schema_name = NULL,
+           column_name = NULL,
+           exact = FALSE) {
+    if(is.null(catalog_name) || is.null(schema_name)) {
+      res <- dbGetQuery(conn, "SELECT CURRENT_DATABASE() AS CAT, CURRENT_SCHEMA() AS SCH")
+      if(is.null(catalog_name) && !is.na(res$CAT[1])) {
+        catalog_name <- res$CAT[1]
```

Comment:
From what I can tell in docs and experimenting locally, `CAT` and `SCH` will always be length one. Any reason why we're doing this `[1]`?